### DISCUSSION TITLE
Problemas con opción de rutina

### DISCUSSION THREAD

**Discussion by Mauricio8086** [olikraus/u8g2#1548] [Q&A]
Buen día , soy nuevo en Arduino, tengo un problema ,algo estoy haciendo mal y no se como solucionarlo, el programa en determinado momento te la opción  Ok o Cancelar y depende lo que elijas debe seguir dicha rutina y no logro que funcione, le paso el código para ver si me pueden ayudar, desde ya gracias!

``` C++
void loop(void) {
  current_selection = u8g2.userInterfaceSelectionList(
                        "M E N U",
                        current_selection,
                        string_list);

  if ( current_selection == 0 ) {
    u8g2.userInterfaceMessage(
      "Nada Seleccionado.",
      "",
      "",
      " ok ");
  } else {
    u8g2.userInterfaceMessage(
      "Selection:",
      u8x8_GetStringLineStart(current_selection - 1 , string_list ),
      ""," Ok \n Cancel ");

    if (current_selection == 1) {         // Ok
      pantalla_OK(); 
    }
else
    if (current_selection == 2) {         // Cancel
      pantalla_CANCEL();
    }
  }
} 
```

**Comment by olikraus** [comment]
google translate: 
> Good morning, I am new to Arduino, I have a problem, I am doing something wrong and I do not know how to solve it, the program at a certain moment gives you the option Ok or Cancel and it depends on what you choose, it must follow this routine and I cannot make it work, I will the code to see if you can help me, thanks in advance! 

You probably want to check for `current_selection == 1` and `current_selection == 2` instead of `current_selection == 0`, see also
https://github.com/olikraus/u8g2/wiki/u8g2reference#userinterfaceselectionlist

**Reply by Mauricio8086** [reply]
Hola buen dia, gracias por  la ayuda, realmente lo que no se como capturar el valor que devuelve 1 an para si se ha seleccionado uno de los botones. 0 si el usuario ha presionado el botón de inicio / cancelar, probe con if y con switch case y no me funciona no logro encontrar un ejemplo que me sirva de ayuda,
Ademas hay una referencia donde uint8_t u8g2_UserInterfaceMessage (u8g2_t * u8g2, const char * title1, const char * title2, const char * title3, const char * botones);
y realmente no se como tomar el valor de * bottons , la variable 0 y 1 para utilizar en con if else o switch case
